x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/TransportStopDataFrameTransformAction.java

```diff
@@ -88,8 +91,31 @@ protected void doExecute(Task task, StopDataFrameTransformAction.Request request
                 new PageParams(0, 10_000),
                 request.isAllowNoMatch(),
                 ActionListener.wrap(hitsAndIds -> {
+                    if (request.isForce() == false) {
+                        PersistentTasksCustomMetaData tasks = state.metaData().custom(PersistentTasksCustomMetaData.TYPE);
```

Comment:
This logic is the main change. If you really want to test this code extract it to a static method and it is easily unit testable. `MlTasksTests` is a good reference for building `PersistentTasksCustomMetaData`